components/dashboard/src/components/ContextMenu.tsx

```diff
@@ -56,11 +56,11 @@ function ContextMenu(props: ContextMenuProps) {
 
     useEffect(() => {
         window.addEventListener("keydown", handler);
-        window.addEventListener("click", clickHandler);
+        window.addEventListener("click", clickHandler, { capture: true });
         // Remove event listeners on cleanup
         return () => {
             window.removeEventListener("keydown", handler);
-            window.removeEventListener("click", clickHandler);
+            window.removeEventListener("click", clickHandler, { capture: true });
```

Comment:
This seems to fix context menus in modals, but has a weird side-effect that you need to click twice to hide a menu again. 🤔

I wonder if this renders the "only skip once" trick obsolete.

Reply:
Actually that doesn't really work.

The problem is that the Modal modal calls `e.stopPropagation()` on click events here:

https://github.com/gitpod-io/gitpod/blob/924b20eaee81481451a163f996cf7d7a8e4b51ab/components/dashboard/src/components/Modal.tsx#L82

This prevent any clicks inside the Modal window from propagating up to the `window` and thus from closing the ContextMenu when needed, because this handler never gets called:

https://github.com/gitpod-io/gitpod/blob/924b20eaee81481451a163f996cf7d7a8e4b51ab/components/dashboard/src/components/ContextMenu.tsx#L59